src/coreclr/vm/jitinterface.cpp

```diff
@@ -11950,10 +11954,31 @@ void* CEEJitInfo::getFieldAddress(CORINFO_FIELD_HANDLE fieldHnd,
 
         GCX_COOP();
 
-        base = (void *) field->GetBase();
-    }
+        base = (void*)field->GetBase();
+        _ASSERTE(base != nullptr);
 
-    result = field->GetStaticAddressHandle(base);
+        result = field->GetStaticAddressHandle(base);
+        _ASSERTE(result != nullptr);
+
+        if (pBoxedStatic != NULL && field->IsStatic() && !field->IsSpecialStatic() && field->IsByValue())
+        {
+            // Check if the field holds a frozen boxed static, return its content in that case
+            MethodTable* fieldType = field->GetFieldTypeHandleThrowing().GetMethodTable();
+            if (!pMT->IsSharedByGenericInstantiations() && !fieldType->ContainsPointers())
+            {
+                Object* boxedStatic = *(Object**)result;
+                if (boxedStatic != nullptr && GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(boxedStatic))
```

Comment:
```suggestion
                if (boxedStatic != nullptr && !boxedStatic->GetMethodTable()->ContainsPointers() && GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(boxedStatic))
```
This is much cheaper than hitting the type loader to get the field type.